src/v0/destinations/discord/util.js

```diff
@@ -0,0 +1,103 @@
+/* eslint-disable no-nested-ternary */
+const { getFieldValueFromMessage } = require("../../util");
+
+/**
+ * get the name value from following heirarchy
+ * traits.name
+ * traits.firstName+traits.lastName (or either)
+ * traits.username
+ * traits.email
+ * properties.email
+ * userId
+ * anonymousId
+ * @param {*} message
+ * @returns uname
+ */
+const getName = message => {
+  const traits = getFieldValueFromMessage(message, "traits");
+  let uName;
+  if (traits) {
+    uName =
+      traits.name ||
+      (traits.firstName
+        ? traits.lastName
+          ? `${traits.firstName}${traits.lastName}`
+          : traits.firstName
+        : undefined) ||
+      traits.username ||
+      (message.properties ? message.properties.email : undefined) ||
+      traits.email ||
+      (message.userId ? `User ${message.userId}` : undefined) ||
+      `Anonymous user ${message.anonymousId}`;
+  } else {
+    uName =
+      (message.properties ? message.properties.email : undefined) ||
+      (message.userId ? `User ${message.userId}` : undefined) ||
+      `Anonymous user ${message.anonymousId}`;
+  }
+
+  return uName;
+};
+
+/**
+ * To get whitelisted traits from config
+ * @param {*} destination
+ * @param {*} traitsList map the whitelisted traits to this list
+ */
+const getWhiteListedTraits = (destination, traitsList) => {
+  if (destination?.Config?.whitelistedTraitsSettings) {
+    destination.Config.whitelistedTraitsSettings.forEach(whiteListTrait => {
+      if (whiteListTrait.trait) {
+        const tmpWhitelistTrait = whiteListTrait.trait.trim();
+        if (tmpWhitelistTrait.trim().length !== 0) {
+          traitsList.push(tmpWhitelistTrait);
+        }
+      }
+    });
+  }
+};
+
+/**
+ * Stringifying json for traits
+ * Not using JSON.stringify() since we forst want to check for whiteListedTraits
+ * @param {*} json input json to be stringified
+ * @param {*} whiteListedTraits
+ * @returns Stringified Json
+ */
+const stringifyJSON = (json, whiteListedTraits) => {
```

Comment:
Can't we use `JSON.stringify()`?

Reply:
We can not since we are first checking if whielistedTraits are mentioned or not if yes then we are stringifying only whiteListed ones otherwise all